contracts/modules/TransferManager/VolumeDumpingRestrictionTM.sol

```diff
@@ -0,0 +1,366 @@
+    * @param _startTime When the dumping restriction, 0 means now in seconds
+    * @param _endTime When the dumping restriction ends in seconds
+    * @param _rollingPeriod Ttime period in seconds
+    */
+    function addDumpingRestriction(
+        address _userAddress, 
+        uint256 _percentAllowed, 
+        uint256 _startTime, 
+        uint256 _endTime, 
+        uint256 _rollingPeriod ) public withPerm(ADMIN) {
+
+        // if a startTime of 0 is passed in, then start now.
+        if(_startTime == 0) {
+            _startTime = now;
+        }
+
+        _checkVolumeDumpingParams(_userAddress, _percentAllowed, _startTime, _endTime, _rollingPeriod);
+        
+        // deny if restriction already exists
+        require(volumeRestriction[_userAddress].percentAllowed == 0, "volume dumping restriction already exists");
+        
+        // add the volume dumping restriction
+        volumeRestriction[_userAddress] = VolumeRestriction(_percentAllowed, _startTime, _endTime, _rollingPeriod);
+
+        emit AddNewVolumeDumping(
+            _userAddress,
+            _percentAllowed,
+            _startTime,
+            _endTime,
+            _rollingPeriod
+        );
+    }
+
+    /**
+    * @notice lets the admin create multiple volume dumping restriction for a multiple addresses.
+    * @param _userAddresses Address of the user to apply the volume dumping restriction
+    * @param _percentsAllowed Percent of tokens balance allowed to transfer within a rolling period
+    * @param _startTimes When the dumping restriction, 0 means now in seconds
+    * @param _endTimes When the dumping restriction ends in seconds
+    * @param _rollingPeriods Time period in seconds
+    */
+    function addDumpingRestrictionMulti(
+        address[] _userAddresses,
+        uint256[] _percentsAllowed, 
+        uint256[] _startTimes, 
+        uint256[] _endTimes, 
+        uint256[] _rollingPeriods) public withPerm(ADMIN) {
+        require( 
+            _userAddresses.length == _percentsAllowed.length &&
+            _userAddresses.length == _rollingPeriods.length &&
+            _userAddresses.length == _startTimes.length &&
+            _userAddresses.length == _endTimes.length,
+            "Input array length mis-match"
+        );
+
+        for (uint256 i = 0; i < _userAddresses.length; i++) {
+            addDumpingRestriction(_userAddresses[i], _percentsAllowed[i], _startTimes[i], _endTimes[i], _rollingPeriods[i]);
+        }
+    }
+
+    /**
+    * @notice lets the admin remove a volume dumping restriction.
+    * @param _userAddress User address to remove the volume dumping restriction
+    */
+    function removeRestriction(address _userAddress) public withPerm(ADMIN) {
+        VolumeRestriction memory toRemove = volumeRestriction[_userAddress];
+
+        // check if entry is in the past
+        // not allowed to delete past entries
+        require(toRemove.endTime > now, "Cannot remove past restrictions");
```

Comment:
It doesn't make sense to delete the past entries. If issuer accidentally add someone then no way to remove the restriction.

Reply:
`If issuer accidentally add someone then no way to remove the restriction.` As far has the endTime hasn't been exceeded the admin can remove the restriction.

Reply:
Sorry My bad, I misread